### PR TITLE
Changes to support the HDFS in Power PC.

### PR DESCRIPTION
**Pull request by MeenaRenganathan22**
Enabling the HDFS support in PowerPC.

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable the HDFS support in PowerPC and which helps to fixes the following functional tests 02113_hdfs_assert.sh, 02244_hdfs_cluster.sql and 02368_cancel_write_into_hdfs.sh.

### Documentation entry for user-facing changes

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by MeenaRenganathan22** [comment]
Related PRs in submodule libhdfs3: https://github.com/ClickHouse/libhdfs3/pull/34
libgsasl: https://github.com/ClickHouse/libgsasl/pull/8

**Comment by kssenii** [comment]
Need to update submodules manually to have the latest commits (pull the branch and commit here), because they are not automatically updated when you merge with master.

**Comment by MeenaRenganathan22** [comment]
> Need to update submodules manually to have the latest commits (pull the branch and commit here), because they are not automatically updated when you merge with master.

Updated the submodules.

**Comment by MeenaRenganathan22** [comment]
@kssenii,

New libhdfs3 submodule PR [35](https://github.com/ClickHouse/libhdfs3/pull/35) been raised due to the IntelAsmCrc32c.cpp got missed in the CMakeLists.txt. 
This change is not part of this PR. Since the libhdfs3 submodule has not been updated to the latest commit, this was not noticed earlier.

Also, please suggest how to include the crc_iscsi_v_pcl.asm file. Not sure whether it needs any special properties in compilation.

**Comment by kssenii** [comment]
> New libhdfs3 submodule PR https://github.com/ClickHouse/libhdfs3/pull/35 been raised due to the IntelAsmCrc32c.cpp got missed in the CMakeLists.txt.

Merged it.

> Also, please suggest how to include the crc_iscsi_v_pcl.asm file. Not sure whether it needs any special properties in compilation.

check `base/glibc-compatibility/`, `CMakeLists.txt` there.

**Comment by MeenaRenganathan22** [comment]
> > New libhdfs3 submodule PR [ClickHouse/libhdfs3#35](https://github.com/ClickHouse/libhdfs3/pull/35) been raised due to the IntelAsmCrc32c.cpp got missed in the CMakeLists.txt.
> 
> Merged it.
> 
> > Also, please suggest how to include the crc_iscsi_v_pcl.asm file. Not sure whether it needs any special properties in compilation.
> 
> check `base/glibc-compatibility/`, `CMakeLists.txt` there.

@kssenii,
Done the changes. Could you please approve to make the workflow get started.

**Comment by kssenii** [comment]
the build is OK
Screenshot 2023-01-27 at 14 22 58
though some hdfs tests are failing, let's rerun the tests, not sure if related

**Comment by MeenaRenganathan22** [comment]
@kssenii,

There are two PRs in libhdfs3 submodule [21](https://github.com/ClickHouse/libhdfs3/pull/21) and [24](https://github.com/ClickHouse/libhdfs3/pull/24) which brings lot of changes to the submodule. But the associated ClickHouse PR [39252](https://github.com/ClickHouse/ClickHouse/pull/39252) has been close abruptly.

 ClickHouse's submodule(libhdfs3 and libgsasl) has not been updated to the latest commit for long time. Now, in this PR, after the submodule has been updated to the latest commit, we 're seeing issue in the functional and integration test related with HDFS.

The above two PRs brings lot of changes and it's very difficult to understand the changes and fix the issue.

Could you please provide your suggestion how to handle the HDFS test failure observing in the PR?

**Comment by kssenii** [comment]
Could you please locally revert those PR's in libhdfs, push it to your own repository and make the submodules reference the branch from your repository and push to this PR so that we test it in CI? I agree that you are right and those submodules probably affected it. But better to test before reverting them in there to be sure.

**Comment by MeenaRenganathan22** [comment]
> Could you please locally revert those PR's in libhdfs, push it to your own repository and make the submodules reference the branch from your repository and push to this PR so that we test it in CI? I agree that you are right and those submodules probably affected it. But better to test before reverting them in there to be sure.



It's done. Can you please approve to test the changes?

**Comment by kssenii** [comment]
@MeenaRenganathan22, I reverted this one https://github.com/ClickHouse/libhdfs3/pull/40, let's update. Not sure if this one https://github.com/ClickHouse/libhdfs3/pull/21 also breaks the tests, will revert if it will still be broken.

**Comment by MeenaRenganathan22** [comment]
> @MeenaRenganathan22, I reverted this one [ClickHouse/libhdfs3#40](https://github.com/ClickHouse/libhdfs3/pull/40), let's update. Not sure if this one [ClickHouse/libhdfs3#21](https://github.com/ClickHouse/libhdfs3/pull/21) also breaks the tests, will revert if it will still be broken.

@kssenii,

Looks like the HDFS tests are failing in Integration and Functional test.
When I revert the PRs 21 and 24 in my branch, I don't see the same failures. 

So, we have to revert the PR 21 as well.

**Comment by kssenii** [comment]
@MeenaRenganathan22, done - https://github.com/ClickHouse/libhdfs3/pull/41.

**Comment by MeenaRenganathan22** [comment]
> @MeenaRenganathan22, done - [ClickHouse/libhdfs3#41](https://github.com/ClickHouse/libhdfs3/pull/41).

Updated the submodule to reflect the latest changes.

**Review by kssenii — APPROVED** [review]

File: docker/server/Dockerfile.ubuntu [hunk 1]
```diff
@@ -16,6 +16,7 @@ RUN sed -i "s|http://archive.ubuntu.com|${apt_archive}|g" /etc/apt/sources.list
         gnupg \
         locales \
         wget \
+        yasm \
```
Comment (tavplubix) [inline docker/server/Dockerfile.ubuntu:19]: Why do we need it here?

Comment (MeenaRenganathan22) [reply docker/server/Dockerfile.ubuntu:19]: By default, yasm is not installed in ubuntu.  In libhdfs3 submodule, this PR (https://github.com/ClickHouse/libhdfs3/pull/22) introduced the file [src/common/crc_iscsi_v_pcl.asm]. 
That time they only updated the submodule and not the ClickHouse submodule reference.

When the ClickHouse's submodule has been updated to the latest commit, there was build error. To support the PR changes, the docker file has been updated to install yasm.

Comment (MeenaRenganathan22) [reply docker/server/Dockerfile.ubuntu:19]: Ok, the PR (https://github.com/ClickHouse/ClickHouse/pull/37949) raised already to support the changes but it has been merged. We used the PR as reference, as it already has been reviewed.

Comment (tavplubix) [reply docker/server/Dockerfile.ubuntu:19]: It does not answer the question. Let's try to remove it and see what will happen - https://github.com/ClickHouse/ClickHouse/pull/46173

**Pull request merged**